### ISSUE TITLE
Weird things in VideoBaseTexture

### ISSUE THREAD

**Issue by jeremypetrequin** [pixijs/pixi.js#2687]
Hi everyone,

Working with VideoBaseTexture lately, I've noticed several small things in the source code that doesn't seem to be "normal" to me.
I had to fix some things directly in the code to have my stuffs working nicely (but not sur if what i've done is good enough to be pushed, I prefer talk to you about it ;) )

So, here's my thoughts:
First, the video automatically play.. I don't think the texture has to control the video, the control has to be handled externaly, and the texture only render it.
https://github.com/pixijs/pixi.js/blob/master/src/core/textures/VideoBaseTexture.js#L132

Secondly, if you load the video before creating the texture, or externaly, the texture doesn't listen for play/pause event, so the texture never update itself (If ``this.complete`` is true)

In constructor, line 60
https://github.com/pixijs/pixi.js/blob/master/src/core/textures/VideoBaseTexture.js#L60
if the video is not loaded, events are added, but if the video is loaded: no events are added. The two events "play/pause" need to be after the ``if(!this.complete)`` no?

And so, if the video is already loaded, the _onCanPlay method is never called, so I think it need a call to it in the constructor.

Last thing, in the _onCanPlay method, in place of the ``this.source.play();``, the texture should make something like that I think:
```javascript
if(!this.source.paused) {
   this._onPlayStart()
}
````

So to sum up, in my mind the end of the constructor should be something like: 
```javascript
    // started playing..
    source.addEventListener('play', this._onPlayStart.bind(this));
    source.addEventListener('pause', this._onPlayStop.bind(this));
    this.__loaded = false;

    if (!source.complete)
    {
        source.addEventListener('canplay', this._onCanPlay);
        source.addEventListener('canplaythrough', this._onCanPlay);
    } else {
        this._onCanPlay();
    }
````

And replace ``this.source.play()`` in method _onCanPlay, by:
```javascript
if(!this.source.paused) {
   this._onPlayStart()
}
````

Sorry for my bad english, I hope you will understand what i mean :)
What do you think about that?

**Comment by englercj** [comment]
Your English is totally fine :) thanks for the bug report, hopefully some can look into it soon. I definitely agree with this:

> I don't think the texture has to control the video, the control has to be handled externaly, and the texture only render it

I'd rather that the texture not have any control over play/pause etc. Just listen to events and update.

**Comment by GoodBoyDigital** [comment]
Hi @jeremypetrequin ! Would you mind submitting your PR again to the dev branch? I think these refinements would really enhance the video class!
